packages/loot-core/src/platform/server/connection/index.electron.ts

```diff
@@ -22,13 +22,23 @@ export const init: T.Init = function (socketName, handlers) {
 
   // websockets doesn't support sending objects so parse/stringify needed
   wss.on('connection', function connection(ws) {
+    ws.on('error', console.error);
+
     ws.on('message', data => {
       let msg = JSON.parse(data);
+
+      if (ws.readyState !== 1) {
+        return;
+      }
+
       let { id, name, args, undoTag, catchErrors } = msg;
 
       if (handlers[name]) {
         runHandler(handlers[name], args, { undoTag, name }).then(
           result => {
+            if (ws.readyState !== 1) {
+              return;
+            }
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏doesn't line 30-32 already cove this?

Reply:
If runHandler executes, takes some undetermined amount of time, and then after perform the `then` statement, sometimes there is a risk that the socket has died midway during the execution of the handler. I encountered this if you for example continuously spam Ctrl+R